### PR TITLE
Project info page

### LINKED ISSUE #212: Project info page

### PR DESCRIPTION
**Pull request by patrickfeltes**
Closes #212 

Notes:
- Waiting on background color for maps (line won't be there)
- There is a horizontal ScrollView for the project tags, but we probably won't every go past screen width because real projects won't have more than 1-2 tags?

Here's a screenshot:
project_info

### CODE REVIEW

**Comment by bstrauch24** [comment]
I think we should put the project number next to the project, i.e. "#1 HackIllinois API"
And yeah, fair to assume projects will not have all four tags

File: app/src/main/java/org/hackillinois/android/database/dao/ProjectsDao.kt [hunk 1]
```diff
@@ -6,6 +6,10 @@ import org.hackillinois.android.database.entity.Project
 
 @Dao
 interface ProjectsDao {
+
+    @Query("SELECT * FROM PROJECTS WHERE id = :projectId LIMIT 1")
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/database/dao/ProjectsDao.kt]: Is the LIMIT 1 necessary?

File: app/src/main/java/org/hackillinois/android/database/entity/Project.kt [hunk 2]
```diff
@@ -18,11 +21,70 @@ data class Project(
...
+        "ECEB_1" to R.drawable.eceb_floor_1,
+        "ECEB_2" to R.drawable.eceb_floor_2,
+        "ECEB_3" to R.drawable.eceb_floor_3,
+        "DCL" to R.drawable.dcl,
+        "Kenney" to R.drawable.kenney
+    )
+
+    fun getBuildingName(): String {
+        val split = room.split(" ")
+        return if (split.isEmpty()) {
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/database/entity/Project.kt]: I would do split.length > 0 here, and make "" the else result

**Comment by patrickfeltes** [comment]
> I think we should put the project number next to the project, i.e. "#1 HackIllinois API"
> And yeah, fair to assume projects will not have all four tags

Done

File: app/src/main/java/org/hackillinois/android/database/entity/Project.kt [hunk 3]
```diff
@@ -18,11 +21,70 @@ data class Project(
...
+    private val NO_RESOURCE = 0
+
+    fun getIndoorMapResource(): Int {
+        val split = room.split(" ")
+        if (split.isEmpty()) {
+            return NO_RESOURCE
+        }
+
+        var buildingKey = split[0]
+        if (split.size >= 2) {
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/database/entity/Project.kt]: Siebel Atrium is also a valid location... I guess if it starts with "Siebel", look for a 0, then look for a 2, else default to 1st floor. Same thing with ECEB Atrium.

File: app/src/main/java/org/hackillinois/android/database/entity/Project.kt (hunk 2 above)
Comment (bstrauch24) [reply app/src/main/java/org/hackillinois/android/database/entity/Project.kt]: Better to make the most common case the first thing in an if/else

File: app/src/main/java/org/hackillinois/android/database/entity/Project.kt (hunk 3 above)
Comment (patrickfeltes) [reply app/src/main/java/org/hackillinois/android/database/entity/Project.kt]: Will there ever be projects located in Kenney or DCL? Should I just remove those?

Comment (patrickfeltes) [reply app/src/main/java/org/hackillinois/android/database/entity/Project.kt]: I removed them, lmk if it should be different

File: app/src/main/java/org/hackillinois/android/database/entity/Project.kt [hunk 4]
```diff
@@ -59,32 +47,44 @@ data class Project(
             return NO_RESOURCE
         }
 
-        var buildingKey = split[0]
-        if (split.size >= 2) {
+        val buildingName = split[0]
+
+        return if (buildingName == "Siebel") {
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/database/entity/Project.kt]: Why don't we have DCL and Kenney anymore?

Comment (patrickfeltes) [reply app/src/main/java/org/hackillinois/android/database/entity/Project.kt]: Wrote it in a different comment, but why do we need them for mentors? Will they ever have a room in either building?

Comment (bstrauch24) [reply app/src/main/java/org/hackillinois/android/database/entity/Project.kt]: Ohhh true. I feel dumb now. Approving.

**Pull request merged**